frontend/tests/unit/tests/work_packages/directives/inplace_editor/custom/editable/inplace-editor-dropdown-directive-test.js

```diff
@@ -0,0 +1,132 @@
+//-- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+describe.only('Inplace editor dropdown directive', function() {
+  var element,
+      scope,
+      html,
+      workPackageFieldService = {},
+      workPackageFieldConfigurationService = {},
+      allowedValues = [],
+      angularCompile;
+
+  html = '<div><inplace-editor-dropdown></inplace-editor-dropdown></div>';
+
+  beforeEach(angular.mock.module('openproject.workPackages.directives'));
+
+  beforeEach(module('openproject.services', function($provide) {
+    $provide.constant('WorkPackageFieldService', workPackageFieldService);
+    $provide.constant('WorkPackageService', {});
+    $provide.constant('WorkPackageFieldConfigurationService', workPackageFieldConfigurationService);
+  }));
+
+  beforeEach(module('openproject.templates'));
+
+  beforeEach(inject(function($rootScope, $compile, $q) {
+    angularCompile = $compile;
+
+    scope = $rootScope.$new();
+
+    allowedValues = [{
+                       href: '/1',
+                       name: 'zzzzzz'
+                     },
+                     {
+                       href: '/2',
+                       name: 'mmmmmm'
+                     },
+                     {
+                       href: '/3',
+                       name: 'aaaaaa'
+                     }];
+
+    var allowedValuePromise = $q(function(resolve) {
+      resolve(allowedValues);
+    });
+
+    workPackageFieldService.getAllowedValues = sinon.stub().returns(allowedValuePromise);
+    workPackageFieldService.isRequired = sinon.stub().returns(true);
+    workPackageFieldConfigurationService.getDropdownSortingStrategy = sinon.stub().returns('');
```

Comment:
`''` is a return value that will never occur from the real implementation.
The real implementation might return `null` or a field name.